### ISSUE TITLE
error: no suitable path for sector found

### ISSUE THREAD

**Issue by SmarterEric** [filecoin-project/lotus#1262]
During seal-worker transferring Cache data to miner, there always an error happened "no suitable path for sector found". 
![image](https://user-images.githubusercontent.com/38749449/74397175-09adf600-4e4f-11ea-8af8-c6e14d560548.png)
the error came from this function when it's called.
```
func (f *FS) findBestPath(size uint64, cache bool, strict bool) (StoragePath, error)
...
if best == "" {
 if cache {
  return best, xerrors.Errorf("no available cache: %w", ErrNoSuitablePath)
 }
```
```
	path, err := sm.SectorBuilder.SectorPath(fs.DataType(vars["type"]), id)
	if err != nil {
		if err != fs.ErrNotFound {
			log.Error(err)
			w.WriteHeader(500)
			return
		}
// during going through this function, and got the error
		path, err = sm.SectorBuilder.AllocSectorPath(fs.DataType(vars["type"]), id, true)
		if err != nil {
			log.Error(err)
			w.WriteHeader(500)
			return
		}
```
I have checked the free disk space using `df -h`, return over 6T space, which was enough.
then check the ~/.lotusstorage/config.toml
![image](https://user-images.githubusercontent.com/38749449/74397755-63fb8680-4e50-11ea-8276-3daf719d0c8d.png)
- config file:
```
# Default config:
[API]
  ListenAddress = "/ip4/192.168.10.31/tcp/2345/http"
  Timeout = "30s"
#
[Libp2p]
  ListenAddresses = ["/ip4/192.168.10.0/tcp/0", "/ip6/::/tcp/0"]
  ConnMgrLow = 150
  ConnMgrHigh = 180
  ConnMgrGrace = "20s"
#
[SectorBuilder]
#  Path = ""
  WorkerCount = 3
  DisableLocalPreCommit = false
  DisableLocalCommit = false
#
```
- error content
```
2020-02-12T09:06:01.542+0800 INFO node impl/storminer.go:50 SERVEGETREMOTE /remote/cache/82
2020-02-12T09:06:01.542+0800 ERROR node impl/storminer.go:122 no available cache: no suitable path for sector fond
github.com/filecoin-project/lotus/node/impl.(*StorageMinerAPI).remotePutSector
 /golang/src/github.com/filecoin-project/lotus/node/impl/storminer.go:122
net/http.HandlerFunc.ServeHTTP
 /usr/local/go/src/net/http/server.go:2007
github.com/gorilla/mux.(*Router).ServeHTTP
 /golang/pkg/mod/github.com/gorilla/mux@v1.7.3/mux.go:212
github.com/filecoin-project/lotus/node/impl.(*StorageMinerAPI).ServeRemote
 /golang/src/github.com/filecoin-project/lotus/node/impl/storminer.go:52
net/http.HandlerFunc.ServeHTTP
 /usr/local/go/src/net/http/server.go:2007
github.com/gorilla/mux.(*Router).ServeHTTP
 /golang/pkg/mod/github.com/gorilla/mux@v1.7.3/mux.go:212
github.com/filecoin-project/lotus/lib/auth.(*Handler).ServeHTTP
 /golang/src/github.com/filecoin-project/lotus/lib/auth/handler.go:50
net/http.serverHandler.ServeHTTP
 /usr/local/go/src/net/http/server.go:2802
net/http.(*conn).serve
 /usr/local/go/src/net/http/server.go:1890
```

**Comment by travisperson** [comment]
Did you check the logs on the storage miner? If the `availableBytes` call fails it will return no path, but an error log should be printed out.

**Comment by SmarterEric** [comment]
> Did you check the logs on the storage miner? If the `availableBytes` call fails it will return no path, but an error log should be printed out.

maybe `availableBytes ` call failed, but the miner has over 6 T storage space left. Also this call is the content in `findBestPath` body, so the error printed out was `no available cache: no suitable path for sector fond`

**Comment by travisperson** [comment]
The call is happening over RPC from the seal worker, so the error returned is `no available cache: no suitable path for sector fond`, but on the storage miner itself where the code is executed if `availableBytes` fails it will just print an error log. The error will not be returned.

There are only three direct failure cases:
1) No configured cache dir
    This is very unlikely as given the configuration provided, the storage miner will always be configured with the cache enabled.
2) There is not enough available space
    Because you say that you have 6 TB of storage free, we can assume this is not the case either
3) `availableBytes` fails for some reason.
   This only leaves this case.

Knowing if the call to `availableBytes` failed on the storage miner would help determine what is actually wrong.

I do believe that the errors returned from the sector builder should be more detailed though.

**Comment by Tony-1240645041** [comment]
This a warm,not error.You Can ignore it.

**Comment by wangkaikai12345** [comment]
can you tell me How to operate normally

**Comment by wangkaikai12345** [comment]
@SmarterEric

**Comment by magik6k** [comment]
Fixed in #1924
